src/coreclr/src/gc/unix/cgroup.cpp

```diff
@@ -431,23 +621,23 @@ class CGroup
         if (file == nullptr)
             goto done;
 
-        if (getline(&line, &lineLen, file) == -1)
+        if (fgets(line, lineLen, file) == nullptr)
```

Comment:
I'd prefer using the getline everywhere unless there is a measurable perf difference.

Reply:
This (among other things) was suggested in the previous review here: https://github.com/dotnet/coreclr/pull/25906

Reply:
I am sorry I've missed that. My general attitude to coding is readability first and optimizing where it matters. Code that uses higher level primitives is usually easier to understand and so bugs are also easier to spot.

Reply:
Fair enough. I will fix up everything that you request. Just please bear with me because it's going to be a bit slow.